src/components/sbb-toggle-check/sbb-toggle-check.tsx

```diff
@@ -47,12 +47,26 @@ export class SbbToggleCheck implements AccessibilityProperties {
 
   @Element() private _element!: HTMLElement;
 
+  @State() private _hasLabelText = false;
+
   /** Method triggered on toggle change. */
   public checkedChanged(event: Event): void {
     this.checked = this._checkbox?.checked;
     forwardEventToHost(event, this._element);
   }
 
+  public connectedCallback(): void {
```

Comment:
Can we not use our default approach here with namedSlots? (e.g. in sbb-button, there for the icon)

Reply:
I tried but the label slot is unnamed, so I can't use it (createNamedSlotState requires an input string, hasNamedSlotElement looks for `slot="${name}"` and so on).
I could update the slot in the toggle-check to `<slot name='label' />`, but it means that consumers should set the label like `<span slot=label'>My label</span>` and I don't really like it.
The _hasLabelText approach is taken and renamed from the sbb-button (_hasText).

Reply:
@kyubisation would it be possible to extend your script to listen to an unnamed slot?

Reply:
I would recommend to just render the slot and apply `hidden` to the containing span, depending on whether there is content or not.
While I think it's important to be efficient, it also adds unneeded complexity in this case (at least to me).

Reply:
Sometimes I forget the simple solutions, I agree with lukas